cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -103,11 +107,111 @@
   private final String instanceName;
   private final String sentryAdminGroup;
 
-  AuthBinding(String sentrySite, String instanceName, String sentryAdminGroup) {
+  private final LoadingCache<Principal, Set<String>> groupCache;
```

Comment:
Do we really need a cache for user to group ? I don't think it's an expensive call. Furthermore looking into the getGroups implementation I see that it already has a cache internally which it uses first. Adding another cache here will just cause overhead and can lead to inconsistencies. Any specific reason to add it here ?